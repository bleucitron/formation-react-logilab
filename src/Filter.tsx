import React from 'react';

interface FilterProps {
  text: string;
  filter(): void;
}

export default function Filter({ text, filter }: FilterProps) {
  return <button onClick={filter}>{text}</button>;
}
