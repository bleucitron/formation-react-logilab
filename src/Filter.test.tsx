import React from 'react';
import { shallow, mount } from 'enzyme';
import Filter from './Filter';

describe('<Filter>', () => {
  it('renders without crashing', () => {
    const cb = jest.fn();
    shallow(<Filter text='Filter' filter={cb} />);
  });

  it('matches shallow snapshot', () => {
    const cb = jest.fn();
    const wrapper = shallow(<Filter text='Filter' filter={cb} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('matches complete snapshot', () => {
    const cb = jest.fn();
    const wrapper = mount(<Filter text='Filter' filter={cb} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls filter cb on click', () => {
    const cb = jest.fn();
    const wrapper = shallow(<Filter text='Filter' filter={cb} />);

    wrapper.simulate('click');

    expect(cb).toHaveBeenCalledTimes(1);
  });
});
