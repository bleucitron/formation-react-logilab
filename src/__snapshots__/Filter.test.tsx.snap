// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter> matches complete snapshot 1`] = `
<Filter
  filter={[MockFunction]}
  text="Filter"
>
  <button
    onClick={[MockFunction]}
  >
    Filter
  </button>
</Filter>
`;

exports[`<Filter> matches shallow snapshot 1`] = `
<button
  onClick={[MockFunction]}
>
  Filter
</button>
`;
